uniffi_bindgen/src/bindings/kotlin/gen_kotlin/mod.rs

```diff
@@ -346,33 +346,24 @@ pub mod filters {
         Ok(codetype.canonical_name(oracle()))
     }
 
-    pub fn lower_var(
-        nm: &dyn fmt::Display,
-        codetype: &impl CodeType,
-    ) -> Result<String, askama::Error> {
-        Ok(codetype.lower(oracle(), nm))
+    pub fn ffi_converter_name(codetype: &impl CodeType) -> Result<String, askama::Error> {
+        Ok(codetype.ffi_converter_name(oracle()))
     }
 
-    pub fn write_var(
-        nm: &dyn fmt::Display,
-        target: &dyn fmt::Display,
-        codetype: &impl CodeType,
-    ) -> Result<String, askama::Error> {
-        Ok(codetype.write(oracle(), nm, target))
+    pub fn lower_fn(codetype: &impl CodeType) -> Result<String, askama::Error> {
+        Ok(format!("{}.lower", codetype.ffi_converter_name(oracle())))
     }
 
-    pub fn lift_var(
-        nm: &dyn fmt::Display,
-        codetype: &impl CodeType,
-    ) -> Result<String, askama::Error> {
-        Ok(codetype.lift(oracle(), nm))
+    pub fn write_fn(codetype: &impl CodeType) -> Result<String, askama::Error> {
+        Ok(format!("{}.write", codetype.ffi_converter_name(oracle())))
     }
 
-    pub fn read_var(
-        nm: &dyn fmt::Display,
-        codetype: &impl CodeType,
-    ) -> Result<String, askama::Error> {
-        Ok(codetype.read(oracle(), nm))
+    pub fn lift_fn(codetype: &impl CodeType) -> Result<String, askama::Error> {
+        Ok(format!("{}.lift", codetype.ffi_converter_name(oracle())))
+    }
+
+    pub fn read_fn(codetype: &impl CodeType) -> Result<String, askama::Error> {
+        Ok(format!("{}.read", codetype.ffi_converter_name(oracle())))
```

Comment:
I don't especially like this change from `*_fn` to `*_var`: I'd prefer it if these calls always produced a valid expression within the language it's generating.

Reply:
I don't feel that strongly about this, but one nice thing about the `*_fn` version is that it avoids creating temporary variables in [some cases](https://github.com/mozilla/uniffi-rs/pull/1144/files#diff-f564b605a6ba32d19bcb47128dc9c5c2f01169e2a41fa60ffaaf0b6eea13845c).  I think that was the only time it happened for this refactored, because of the way the Kotlin code uses scope functions.  But there were several instances of that in the scaffolding code that we cleaned up.  I think there will be more instances in the other languages.

That said, I don't think it would be hard to change my mind on this one.  What's the benefits for having the template code generate the full expression?

Reply:
I'm not sure I care about a temporary variable.

Working with expressions provides a more high-level tool to work with the source code than strings, and gives us the flexibility to change the expression for each type.

Reply:
Yes, but I'm not sure that kind of flexibility is good.  I want the FFI functions to be standardized as much as possible for reasons listed at the top.  Can you envision any reason where this system wouldn't work and we would have to use a different kind of expression?

Reply:
In other words, I want whatever complexity is needed to exist inside a Kotlin function rather than a filter function.